### PR TITLE
Update authentication section for clarity

### PR DESCRIPTION
**Pull request by SirSpidey**
Addresses changes requested from reviewers to make the authentication section clearer as we migrate services to IAM.

### Summary

- Update intro to authentication
- Move IAM to top of the stack of auth methods. Rename to just `IAM`
- Update Getting credentials section
- Update Visual Recognition note
- Change headings to sentence case
- Add toc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,68 +65,34 @@ For more details see [#405](https://github.com/watson-developer-cloud/python-sdk
...
-# In the constructor
-discovery = DiscoveryV1(version='2017-10-16', username='<username>', password='<password>')
-```
-
-```python
-# After instantiation
-discovery = DiscoveryV1(version='2017-10-16')
-discovery.set_username_and_password('<username>', '<password>')
-```
+**Tip**: To find out which authentication to use, view the service credentials by clicking the service instance on the IBM Cloud [Dashboard](https://console.bluemix.net/dashboard/apps?watson).
```
Comment (ehdsouza) [inline README.md]: https://console.bluemix.net/dashboard/apps?category=watson

File: README.md [hunk 2]
```diff
@@ -43,68 +65,34 @@ For more details see [#405](https://github.com/watson-developer-cloud/python-sdk
...
-*Important: Instantiation with API key works only with Visual Recognition service instances created before May 23, 2018. Visual Recognition instances created after May 22 use IAM.*
+### Getting credentials
+You find the service credentials for authentication the same way for all Watson services:
 
-```python
-from watson_developer_cloud import VisualRecognitionV3
-# In the constructor
-visual_recognition = VisualRecognitionV3(version='2018-05-22', api_key='<api_key>')
-```
+1.  Go to the IBM Cloud **[Dashboard](https://console.bluemix.net/dashboard/apps?watson)** page.
```
Comment (ehdsouza) [inline README.md]: https://console.bluemix.net/dashboard/apps?category=watson

File: README.md [hunk 3]
```diff
@@ -43,68 +65,34 @@ For more details see [#405](https://github.com/watson-developer-cloud/python-sdk
...
-### Using IAM
+IBM Cloud is migrating to token-based Identity and Access Management (IAM) authentication. IAM authentication uses a service API key to get an access token that is passed with the call. Access tokens are valid for approximately one hour and must be regenerated.
 
-When authenticating with IAM, you have the option of passing in:
+You supply either an IAM service **API key** or an **access token**:
 
-* the IAM API key and, optionally, the IAM service URL
-* an IAM access token
+- Use the API key to have the SDK manage the lifecycle of the access token. The SDK requests an access token, ensures that the access token is valid, and refreshes it if necessary.
+- Use the access token if you want to manage the lifecycle yourself. Access tokens are valid. For details, see [Authenticating with IAM tokens](https://console.stage1.bluemix.net/docs/services/watson/getting-started-iam.html).
```
Comment (ehdsouza) [inline README.md]: stage1?

File: README.md (hunk 1 above)
Comment (SirSpidey) [reply README.md]: This is the link I got from Sabtain and German that's supposed to work. Did it change? Posted comment in slack

File: README.md (hunk 3 above)
Comment (SirSpidey) [reply README.md]: good catch. fixed in java too.

**Pull request merged**